logstash-core-event-java/src/main/java/com/logstash/RubyJavaObject.java

```diff
@@ -0,0 +1,124 @@
+package com.logstash;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import org.jruby.Ruby;
+import org.jruby.RubyNil;
+import org.jruby.runtime.builtin.IRubyObject;
+
+import java.io.InvalidObjectException;
+import java.io.ObjectStreamException;
+import java.io.Serializable;
+import java.io.ObjectInputStream;
+
+public final class RubyJavaObject implements Serializable {
+    private static final long serialVersionUID = 4358368393114318978L;
+
+    private transient IRubyObject rubyValue;
+    private transient boolean isRubyNil;
+    private Object javaValue;
+
+    public RubyJavaObject(Object javaValue) {
```

Comment:
You are correct. After fixing the Javafier for JavaProxy instances, It occurs to me to have specific BiValues for each non-collection ruby/java pair. If so, we move the Javafier/Rubyfier specifics into these types.
WDYT?
I will make this change and we can discuss in here.